sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnSchemaInformation.java

```diff
@@ -101,7 +107,16 @@ DoFnSchemaInformation withSelectFromSchemaParameter(
                     unbox))
             .build();
 
-    return toBuilder().setElementConverters(converters).build();
+    List<FieldAccessDescriptor> descriptors =
+        ImmutableList.<FieldAccessDescriptor>builder()
+            .addAll(getFieldAccessDescriptors())
+            .add(selectDescriptor)
+            .build();
```

Comment:
Instead of storing a List, just store one FieldAccessDescriptor and use FieldAccessDescriptor.union to merge them.

Reply:
Thanks for pointing that out! I was looking for that method.